src/Mirza/BusinessRegistry/Handlers/Location.hs

```diff
@@ -72,24 +73,16 @@ addLocationQuery  :: ( Member context '[]
                   -> NewLocation
                   -> DB context err Location
 addLocationQuery (AuthUser (BT.UserId uId)) locId geoLocId newLoc = do
-  mbizId <- pg $ runSelectReturningOne $ select $ do
-    user <- all_ (_users businessRegistryDB)
-    guard_ (user_id user ==. val_ uId)
-    pure $ user_biz_id user
-  case mbizId of
-               -- Since the user has authenticated, this should never happen
-    Nothing -> throwing _UnexpectedErrorBRE callStack
-    Just bizId -> do
-      let (loc,geoLoc) = newLocationToLocation locId geoLocId bizId newLoc
-      res <- pg $ runInsertReturningList (_locations businessRegistryDB) $
-                  insertValues [loc]
-      case res of
-        [r] -> do
-
-            _ <- pg $ runInsertReturningList (_geoLocations businessRegistryDB) $
-                 insertValues [geoLoc]
-            pure r
-        _   -> throwing _UnexpectedErrorBRE callStack
+  let bizId = BizId . _sglnCompanyPrefix . newLocGLN $ newLoc
```

Comment:
astacey [14:24]
@sajid.anower ok let me see if what I understand from that code is going on is actually going on
so if I understand correctly you've updated so that we don't use the auth user details for GS1 company prefix, but instead use the GS1 company prefix from the newlocation?
is this correct?

sajid.anower [14:26]
yep exactly

astacey [14:26]
but the permission model in the business registry is that the user can only make changes within there business
so what should happen is that we should check that the gs1 company prefix matches
and they can only add the location if it does
and from your tests each business should have a user
and the user from the appropriate business should add the location

sajid.anower [14:27]
yes, that's true. the logic in that function was incorrect regardless

astacey [14:27]
yeah

sajid.anower [14:27]
and we need this question answered before we can figure out what the correct logic is:
https://github.com/data61/Mirza/issues/306

astacey [14:29]
yeah thats right

sajid.anower [14:29]
at the moment, i understand that it sort of breaks the permission model, but we do not really know what the proper permission model should look like

astacey [14:29]
in the absense of any information I'm going to make some executive decisions for the BR

sajid.anower [14:30]
:thumbsup:

sam.paulways [14:30]
:thumbsup:
:wink:
just make the _right_ one
no pressure

astacey [14:30]
that is you can only operate on actions from your own business
and at the moment we only support 1 user per business
I wish I didn't have to add the second constraint
and we should relax that ASAP
but the implementation is compatable with those constraints as far as I know
and it gives us a reasonable upgrade path to an actual useable system in the future

Reply:
I have added a check that the user's bizId is the same as the business being operated on, and if not, and error `OperationNotPermittedBRE` is thrown :+1: